MonoGame.Framework/Graphics/Texture2D.OpenGL.cs

```diff
@@ -70,73 +70,61 @@ public partial class Texture2D : Texture
         private void PlatformConstruct(int width, int height, bool mipmap, SurfaceFormat format, SurfaceType type, bool shared)
         {
             this.glTarget = TextureTarget.Texture2D;
+            format.GetGLFormat(GraphicsDevice, out glInternalFormat, out glFormat, out glType);
+            // GL texture is generated on first access in PlatformSetData
```

Comment:
`GenerateTextureIfRequired` has been added to `PlatformConstruct`.  Running the unit tests on OpenGL, it gets a `GL_INVALID_VALUE` on the first call to `Texture2D.SetData()` on a simple 64x64 Color texture with no mipmaps.  Spent a while trying to track it down without luck.  Should check if it is happening on develop branch.